extension_azuremonitor.tf

```
@@ -0,0 +1,23 @@
+resource "azurerm_virtual_machine_extension" "microsoftmonitoringagent" {
+  count                      = var.log_analytics_agent != null ? 1 : 0
+  name                       = "MicrosoftMonitoringAgent"
+  virtual_machine_id         = azurerm_linux_virtual_machine.this.id
+  publisher                  = "Microsoft.EnterpriseCloud.Monitoring"
+  type                       = "OmsAgentForLinux"
+  type_handler_version       = "1.16"
+  automatic_upgrade_enabled  = true
+  auto_upgrade_minor_version = true
+
+  settings = <<SETTINGS
+    {
+       "workspaceId" : "${var.log_analytics_agent.workspace_id}"
+    }
+SETTINGS
```

```suggestion
  settings = <<-SETTINGS
  {
     "workspaceId" : "${var.log_analytics_agent.workspace_id}"
  }
  SETTINGS
```
When using <<- instead of << it will ignore leading whitespaces. This allows for a bit cleaner formatting, but has absolutely no effect on the json. 

Alternatively you could use ```jsonencode({"workspaceId" = var.log_analytics_agent.workspace_id})``` to get rid of raw json entirely